wagtailmenus/models/menuitems.py

```diff
@@ -96,6 +96,16 @@ def relative_url(self, site=None):
                 return ''
         return self.link_url + self.url_append
 
+    def get_full_url(self):
+        if self.link_page:
+            try:
+                return self.link_page.full_url + self.url_append
+            except TypeError:
+                return ''
+        return self.link_url + self.url_append
+
+    full_url = property(get_full_url)
```

Comment:
If we accept a request parameter as outlined above, declaring a property method probably isn't necessary